source/app/models/url.rb

```diff
@@ -0,0 +1,45 @@
+require 'active_record'
+require 'uri'
+require 'net/http'
+require_relative '../helpers/urls_helper'
+
+class TargetValidator < ActiveModel::Validator
+  def validate(record)
+    if record.target_link.length == 0
+      record.errors[:target_link] << 'URL cannot be an empty string'
+      return
+    end
+    if !record.target_link.match(/https?:\/\//)
+      record.errors[:target_link] << 'URL has unrecognized scheme'
+      return
+    end
+    begin
+      uri = URI.parse(record.target_link)
+      resp = Net::HTTP.get_response(uri)
+      if resp.code >= 400
+        record.errors[:target_link] << 'URL is not reachable'
+      end
+    rescue URI::InvalidURIError
+      record.errors[:target_link] << 'URL is not valid'
+      return
+    rescue
+      record.errors[:target_link] << 'URL is not reacable'
+    end
+  end
+end
+
+class Url < ActiveRecord::Base
+  before_save   :ensure_short_link
+  before_create :ensure_short_link
+  validates :linkid, uniqueness: true
+#  validates :target_link, presence: true
```

Comment:
Why not use this simpler validation over the verbose version in `TargetValidator`?